### DISCUSSION TITLE
esp: no acceptable proposal found with Networkmanager client on Debian

### DISCUSSION THREAD

**Discussion by Arunmor1510** [strongswan/strongswan#1683] [General]
Hi, I'm trying to use `network-manager-strongswan` on debian (bullseye, version 11), but it cannot establish the connection.

I think this is the most important part in the log from the gateway:
```
01[CFG] received proposals: ESP:AES_GCM_16_128/AES_GCM_16_192/AES_GCM_16_256, ESP:AES_CBC_128/AES_CBC_192/AES_CBC_256/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/HMAC_SHA1_96/AES_XCBC_96/NO_EXT_SEQ
01[CFG] configured proposals: ESP:AES_GCM_16_256/CHACHA20_POLY1305/NO_EXT_SEQ, ESP:AES_GCM_16_128/CHACHA20_POLY1305/NO_EXT_SEQ
01[IKE] no acceptable proposal found
```
Adding `-esn`, `-noesn-esn` or `-esn-noesn` on the gateway doesn't work. If I set `esp_proposals = default` on the gateway, or manually set `esp proposal` to `aes256gcm16` on the debian client, things work again.

I still want to be able to manually specify `esp_proposals` on the gateway and make fewer changes to the client if possible, any ideas?

**Comment by tobiasbrunner** [comment]
That was a bug in 5.9.0+ that was fixed with 5.9.6 (956b25a4cc1848fd0b79f61c9d8065ca32f0ffa3).

> I still want to be able to manually specify `esp_proposals` on the gateway and make fewer changes to the client if possible, any ideas?

If you don't want to change the proposal on the client (see the advanced options in the GUI), then you can't use AEAD algorithms as that proposal is invalid. You'd have to configure a classic proposal on the server (e.g. `aes256-sha256`).

**Reply by Arunmor1510** [reply]
> That was a bug in 5.9.0+ that was fixed with 5.9.6 ([956b25a](https://github.com/strongswan/strongswan/commit/956b25a4cc1848fd0b79f61c9d8065ca32f0ffa3)).

Thank you for the information. Then I need to wait a bit more for the next debian release.
